service/controller/v24/resource/cloudformation/delete.go

```diff
@@ -69,6 +75,55 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 	return nil
 }
 
+func (r *Resource) disableMasterTerminationProtection(ctx context.Context, masterInstanceName string) error {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "disabling master instance termination protection")
+
+	i := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String("tag:Name"),
+				Values: []*string{
+					aws.String(masterInstanceName),
+				},
+			},
+		},
+	}
+	o, err := cc.AWSClient.EC2.DescribeInstances(i)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for _, reservation := range o.Reservations {
+
+		if len(reservation.Instances) != 1 {
+			return microerror.Maskf(masterInstanceExistsError, "expected one master instance, got %d", len(reservation.Instances))
```

Comment:
A bit nitty but the error is not really accurate. What about some generic one like `executionFailedError`?

Reply:
Also, how many reservations are there? We should also check that there is only one of them.

Reply:
Good point. We could return a `tooManyError` in that case.

Reply:
When you want to do that we have to check twice. I found that cumbersome in some other places and just checked that there is one item and returned the generic error.

Reply:
Ack that was the previous approach. Agree it is more cumbersome. Lets go with this check and the generic error.